sdk/formrecognizer/ai-form-recognizer/review/ai-form-recognizer.api.md

```diff
@@ -0,0 +1,1299 @@
+                                readonly type: "number";
+                                readonly description: "Individual price of each item unit";
+                                readonly example: "$999.00";
+                            };
+                            readonly Description: {
+                                readonly type: "string";
+                                readonly description: "Item description";
+                                readonly example: "Room Charge";
+                            };
+                            readonly Date: {
+                                readonly type: "date";
+                                readonly description: "Item date";
+                                readonly example: "27Mar21";
+                            };
+                            readonly Category: {
+                                readonly type: "string";
+                                readonly enum: readonly ["Room", "Meals", "Tax", "Deposit", "Credit", "Other"];
+                                readonly description: "Item category";
+                                readonly example: "Room";
+                            };
+                        };
+                    };
+                };
+            };
+        };
+    };
+};
+
+// @public
+export type ReifyFieldSchema<Schema extends Readonly<FieldSchema>> = Schema extends StructuredStringFieldSchema<infer Type> ? {
+    time: DocumentTimeField;
+    phoneNumber: DocumentPhoneNumberField;
+}[Type] : Schema extends StringLikeFieldSchema<infer Type> ? {
+    string: Schema extends {
+        enum: string[];
+    } ? DocumentStringField & {
+        value?: Schema["enum"][number];
+    } : DocumentStringField;
+    currency: DocumentCurrencyField;
+    countryRegion: DocumentCountryRegionField;
+}[Type] : Schema extends NumberFieldSchema<infer Type> ? {
+    number: DocumentNumberField;
+    integer: DocumentIntegerField;
+}[Type] : Schema extends DateFieldSchema ? DocumentDateField : Schema extends ArrayFieldSchema<infer Item> ? DocumentArrayField<ReifyFieldSchema<Item>> : Schema extends ObjectFieldSchema<infer Properties> ? DocumentObjectField<{
+    [K in Extract<keyof Properties, string> as Uncapitalize<K>]?: ReifyFieldSchema<Properties[K]>;
+}> : never;
+
+// @public
+export type ReifyPrebuiltSchema<Schema extends Readonly<ModelSchema>> = {
+    [DocType in keyof Schema["docTypes"]]: {
+        docType: DocType;
+        fields: {
+            [K in Extract<keyof Schema["docTypes"][DocType]["fieldSchema"], string> as Uncapitalize<K>]?: ReifyFieldSchema<Schema["docTypes"][DocType]["fieldSchema"][K]>;
+        };
+    };
+}[keyof Schema["docTypes"]];
+
+// @public
+export type SelectionMarkState = string;
+
+// @public
+export type StringIndexType = typeof StringIndexType[keyof typeof StringIndexType];
+
+// @public
+export const StringIndexType: {
+    readonly Utf16CodeUnit: "utf16CodeUnit";
+    readonly UnicodeCodePoint: "unicodeCodePoint";
+};
```

Comment:
I remember there are issues with const enums when used in JS but I do not remember the details.

Reply:
This isn't a const enum, just a const.

Reply:
No particular hard reason. The type has the same name as the const, so this allows you to write `StringIndexType.UnicodeCodePoint`, makes it so that the type and its values are entangled in the editor, and just offers I think a slightly better developer experience.

Reply:
That is fine but I am worried about how the type is formulated based on the value here. It is good to maintain tight coupling but it is bad for readability and how it will show up in the docs.

Reply:
I'm curious about what docs look like, but I agree this could be a better approach if it works. Cosmos does something similar.